File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    let m=matrix.length;
    let n=matrix[0].length;

    let row=m-1;
    let col=0;

    while(row>=0 && col<n){
        if(matrix[row][col]==target){
            return true;
        }

        if(matrix[row][col]<target){
            col++;
        }
        else{
            row--;
        }
    }
    return false;
};

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.go
func searchMatrix(matrix [][]int, target int) bool {
    m:=len(matrix)
    n:=len(matrix[0])

    row:=m-1
    col:=0

    for row>=0 && col<n{
        if matrix[row][col]==target{
            return true
        }

        if matrix[row][col]<target{
            col++
        }else{
            row--
        }
    }
    return false
}